Add a temp patch fixing Korean/Chinese characters input in RN 0.55.x (iOS).

mandrigin opened the pull request:
The fix itself is described here: facebook/react-native#19809

The patch:
1. A copy of patched RCTUITextView.m file
2. A pre-action in the "Build" scheme to patch the RN file. See more on
   pre-actions here: https://burcugeneci.wordpress.com/2016/05/25/scripts-we-love-xcode-builds-life-savers/
3. A build step to fail the build if the file wasn't patched.

I don't use diffs or anything there because this patch will be applied before every build, essentially.

--

The follow-up issue for using our own RN fork: https://github.com/status-im/status-react/issues/4850

--

fixes: #4818
fixes: #4829

status: ready <!-- Can be ready or wip -->

mandrigin commented:
@lukaszfryc @annadanchenko I temporarily left only iOS build here, it turns out it is not that easy to set it up to a custom branch.

lukaszfryc commented:
@mandrigin looks like it's working! At least for Chinese and Japanese. Previously, each new letter typed was considered individually. Now they are combined which is the expected result according. 

But, when I test Korean, I don't notice any difference so I'm not sure if it's fixed there. I even tried your test scenario from https://github.com/facebook/react-native/pull/19809 and my result is `ㅍㅏㄹㅏㄴ하늘` instead of `파란하늘`. @mandrigin does it work on your side?

Btw, also good news is that I can't reproduce https://github.com/status-im/status-react/issues/4818 on this build!

mandrigin commented:
@lukaszfryc yeah, I was actually testing it on a Korean keyboard, so both gifs in the PR are for it.
did you follow the exact steps? Which OS was it?

lukaszfryc commented:
@mandrigin I've just tested it again on Korean keyboard and it works. Strange, I think had to type some incorrect character previously that didn't match the word. I tested it again on iOS 11.4, 11.2.6 and compared with 0.9.20 to be sure of how it looks there and all looks fine 👍 

Are we sure that it's not affecting Android and there is no need to test it there?

mandrigin commented:
yeah, the changes are in iOS-specific area, it doesn't even compile these files for Android

lukaszfryc commented:
@mandrigin any other things apart from input / keyboard that maybe affected (on iOS)? If not, let's review and merge

mandrigin commented:
@lukaszfryc nope, the code specifically touches *multi-line* text input on iOS.

lukaszfryc commented:
OK. So this works fine. At least on iOS 11.4, 11.2.6 which I think it's enough to test it.

mandrigin committed:
[#4829] Add a patch that fixes complex input on iOS in RN 0.55.x.

The fix is described here: https://github.com/facebook/react-native/pull/19809

The patch:
1. A copy of patched RCTUITextView.m file
2. A pre-action in the "Build" scheme to patch the RN file. See more on
   pre-actions here: https://burcugeneci.wordpress.com/2016/05/25/scripts-we-love-xcode-builds-life-savers/
3. A build step to fail the build if the file wasn't patched.

Signed-off-by: Igor Mandrigin <i@mandrigin.ru>

tonlen commented:
The single line still has problems

lukaszfryc commented:
@tonlen could you describe the single line issue in more details with steps how to reproduce it? Btw, have you tried to reproduce it on the latest version from app store 0.9.21?
